content/client-lib-development-guide/features.textile

```diff
@@ -508,6 +519,268 @@ h3(#eventemitter). EventEmitter mixin / interface
 
+h2(#state-conditions-and-operations). State conditions and operations
+
+@Connection@ operation effects depending on current @Connnection.state@:
+
+<table id="connection-states-operations">
+
+<tr>
+  <th></th>
+  <th>Initialized</th>
+  <th>Connecting</th>
+  <th>Connected</th>
+  <th>Disconnected</th>
+  <th>Suspended</th>
+  <th>Closing</th>
+  <th>Closed</th>
+  <th>Failed</th>
+</tr>
+
+<tr>
+  <td>@connect@</td>
+
+  <!-- When INITIALIZED -->
+  <td>"RTN11a":#RTN11a</td>
+  <!-- When CONNECTING -->
+  <td>No-op</td>
+  <!-- When CONNECTED -->
+  <td>No-op</td>
+  <!-- When DISCONNECTED -->
+  <td>No-op</td>
+  <!-- When SUSPENDED -->
+  <td>No-op</td>
+  <!-- When CLOSING -->
+  <td>"RTN11b":#RTN11b</td>
+  <!-- When CLOSED -->
+  <td>"RTN11a":#RTN11a</td>
+  <!-- When FAILED -->
+  <td>No-op</td>
+
+</tr>
+
+<tr>
+  <td>@close@</td>
+
+  <!-- When INITIALIZED -->
+  <td>No-op</td>
+  <!-- When CONNECTING -->
+  <td>"RTN12f":#RTN12f</td>
+  <!-- When CONNECTED -->
+  <td>"RTN12a":#RTN12a</td>
+  <!-- When DISCONNECTED -->
+  <td>"RTN12d":#RTN12d</td>
+  <!-- When SUSPENDED -->
+  <td>"RTN12e":#RTN12e</td>
+  <!-- When CLOSING -->
+  <td>No-op</td>
+  <!-- When CLOSED -->
+  <td>No-op</td>
+  <!-- When FAILED -->
+  <td>No-op</td>
+
+</tr>
+
+<tr>
+  <td>@ping@</td>
+
+  <!-- When INITIALIZED -->
+  <td>"RTN13b":#RTN13b</td>
+  <!-- When CONNECTING -->
+  <td>"RTN13c":#RTN13c</td>
+  <!-- When CONNECTED -->
+  <td>"RTN13a":#RTN13a</td>
+  <!-- When DISCONNECTED -->
+  <td>"RTN13c":#RTN13c</td>
+  <!-- When SUSPENDED -->
+  <td>"RTN13b":#RTN13b</td>
+  <!-- When CLOSING -->
+  <td>"RTN13b":#RTN13b</td>
+  <!-- When CLOSED -->
+  <td>"RTN13b":#RTN13b</td>
+  <!-- When FAILED -->
+  <td>"RTN13b":#RTN13b</td>
+
+</tr>
+
+<tr>
+  <td>Channel @attach@</td>
+
+  <!-- When INITIALIZED -->
+  <td>"RTL4h":#RTL4h</td>
+  <!-- When CONNECTING -->
+  <td>"RTL4h":#RTL4h</td>
+  <!-- When CONNECTED -->
+  <td>"See channel table":#channel-states-operations</td>
+  <!-- When DISCONNECTED -->
+  <td>"RTL4h":#RTL4h</td>
+  <!-- When SUSPENDED -->
+  <td>"RTL4b":#RTL4b</td>
+  <!-- When CLOSING -->
+  <td>"RTL4b":#RTL4b</td>
+  <!-- When CLOSED -->
+  <td>"RTL4b":#RTL4b</td>
+  <!-- When FAILED -->
+  <td>"RTL4b":#RTL4b</td>
+
+</tr>
+
+<tr>
+  <td>Channel @detach@</td>
+
+  <!-- When INITIALIZED -->
+  <td>"RTL5h":#RTL5h</td>
+  <!-- When CONNECTING -->
+  <td>"RTL5h":#RTL5h</td>
+  <!-- When CONNECTED -->
+  <td>"See channel table":#channel-states-operations</td>
+  <!-- When DISCONNECTED -->
+  <td>"RTL5h":#RTL5h</td>
+  <!-- When SUSPENDED -->
+  <td>"See channel table":#channel-states-operations</td>
```

Comment:
If the connection is SUSPENDED, the channel is DETACHED (RTL3b) or INITIALIZED, and then RTL5a says that nothing is done. I think this is more coherent that failing the request.